src/docfx/build/link/FileLinkMapBuilder.cs

```diff
@@ -54,4 +62,13 @@ public object Build(PublishModel publishModel)
 
         return new { links };
     }
+
+    private bool IsTrusted(string href, string tagName)
```

Comment:
I feel like the protocol check should happen at the `LinkResolver` layer

Reply:
That's one way to sanitize, but then it would be partially duplicated with what's in place since we are not sanitizing attribute here.

And also, the output links in `.links.json` could come from `XrefResolver` as well, if this is unnecessary, we can put the check before https://github.com/dotnet/docfx/blob/afd9c17016a91275ff76f6d2d62da8c00faea665/src/docfx/build/link/LinkResolver.cs#L111

Reply:
I feel like I should have implemented trusted domains at `LinkResolver` and `XrefResolver` in the first place #7220 . We could consolidate them later.